persistence/trees/trees.go

```diff
@@ -0,0 +1,573 @@
+// package trees maintains a set of sparse merkle trees
```

Comment:
NIT: Please use `mv` in the future. 

Did you make any chances to this logic (do I need to review it) or is it just a mv?

Reply:
> @dylanlott Do you plan on adding any unit tests to this?

In follow up PRs, yes, I was just trying to bring it up to test parity for this PR to keep it smaller.

Reply:
> NIT: Please use `mv` in the future.
> 
> Did you make any chances to this logic (do I need to review it) or is it just a mv?

Yes, you should review it all out of caution, but I attempted to keep changes minimal and separated the querying from the actual update application to make it easier to reason about.

Reply:
Any chances you can share a `git diff` that could help me out with it or point me to where I should review the core changes?

I understand the importance of `you should review it all out of caution` but it's 600 lines of business logic and would really help me out if if I only had to evaluate the It's 600 lines of business logic and would really

Reply:
https://gist.github.com/dylanlott/380848a878a72f45b8c2f15231ddfa22 Here's a gist link to the diff between state.go (previous) and trees.go (refactored). The core of the tree changes are in persistence/trees/trees.go.

The only major difference is that we moved the state.go package off of the persistence context and into its own package, and then separated the query functions from the tree update functions for readability. Otherwise, we use the same structure and approach, and especially kept the same order of trees that was declared since that's a sensitive part of the package.